src/main/java/seedu/moneygowhere/commands/ConsoleCommandSortExpense.java

```diff
@@ -92,12 +97,33 @@ public int compare(Expense expense1, Expense expense2) {
     private Comparator<Expense> sortByAmount = new Comparator<>() {
         @Override
         public int compare(Expense expense1, Expense expense2) {
-            BigDecimal expenseName1 = expense1.getAmount();
-            BigDecimal expenseName2 = expense2.getAmount();
+            BigDecimal expenseAmount1 = expense1.getAmount();
+            BigDecimal expenseAmount2 = expense2.getAmount();
+            if (expenseAmount1.equals(expenseAmount2)) {
+                return sortByAlphabet.compare(expense1, expense2);
+            }
             if (order) {
-                return expenseName1.compareTo(expenseName2);
+                return expenseAmount1.compareTo(expenseAmount2);
             }
-            return expenseName2.compareTo(expenseName1);
+            return expenseAmount2.compareTo(expenseAmount1);
+        }
+    };
+
+    /**
+     * Comparator sorts currencies from A to Z if ascending order, Z to A if descending order.
+     */
+    private Comparator<Expense> sortByCurrency = new Comparator<>() {
+        @Override
+        public int compare(Expense expense1, Expense expense2) {
+            String expenseCurrency1 = expense1.getCurrency();
+            String expenseCurrency2 = expense2.getCurrency();
+            if (expenseCurrency1.equalsIgnoreCase(expenseCurrency2)) {
+                return sortByAmount.compare(expense1, expense2);
+            }
+            if (order) {
+                return expenseCurrency1.compareTo(expenseCurrency2);
+            }
+            return expenseCurrency2.compareTo(expenseCurrency1);
```

Comment:
Would you think its better to move the comparator functions into ExpenseManager.java instead since the ConsoleCommand classes are defined to store command arguments for parsing?

Reply:
Makes sense, i'll move the comparators over.